plugins/external_setup_hook/app/models/external_setup_hook.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ExternalSetupHook < ActiveRecord::Base
+  validates :name, presence: true
+
+  validate :validate_endpoint
+
+  has_many :stage_external_setup_hooks, dependent: :destroy
+  has_many :stages, through: :stage_external_setup_hooks, dependent: :destroy
```

Comment:
yeah we don't want to delete all stages when a hook is deleted :)
```suggestion
  has_many :stages, through: :stage_external_setup_hooks
```